gdal/src/main/scala/geotrellis/gdal/Gdal.scala

```diff
@@ -31,7 +32,16 @@ object GdalException {
 object Gdal {
   gdal.AllRegister()
 
-  def open(path: String): RasterDataSet = {
+  def open(
+    path: String,
+    readingParameters: Option[Map[String, String]] = None
+  ): RasterDataSet = {
+    readingParameters
+      .getOrElse(Map[String, String]())
+      .map { case (k, v) =>
+        gdalJNI.SetConfigOption(k, v)
```

Comment:
I see it like 

```scala
object Gdal {
  //...
  def setConfigOption(key: String, value: String): Unit = gdalJNI.SetConfigOption(k, v)
  def setConfigOptions(map: Map[String, String]): Unit = ....
 // ... etc ...
}
```

Reply:
But tbh, i don't see to much sense in doing it :D

Reply:
Yeah, It should still work before the dataset is opened. That's a good point about the deletion, though, I'm not sure how it matters. I do think that your suggestion of `setConfigOption` and `setConfigOptions` are better ways to pass in the parameters.

The reason why we need to pass in the parameters this way is because there's no way to do it in FE without having GDAL as a dependency.

Reply:
Once FE would depend on geotrellis-gdal it will depend on GDAL.

Reply:
Also answering your next question, it doesnt seem like `gdalJNI.SetConfigOption` refs to any dataset.

Reply:
Oh, you're right. I completely forgot about transitive dependencies. I'll close this pr.